### PR TITLE
ports some emotes and emote shortcuts from goon

### PR DESCRIPTION
**Pull request by cacogen**
:cl: cacogen
add: Flex, raise hand and snap fingers emotes
tweak: Improves dance emote
add: Adds more keyboard shortcuts for emotes 
/:cl:

So I went on Goon last night to better understand the ill-fated GoonSounds PR and noticed not for the first time they have a lot more emote shortcuts than us.

Because emote shortcuts are a fun and quick way to communicate with other players without the need to stop and type, I thought I'd update ours to be on level with theirs. This involved porting a couple of emotes we don't have and adding some of their changes to the dance emote.

skin.dmf (where hotkeys are stored) got a lot of changes simply through resaving it, I guess due to BYOND updates. I can probably go through and remove most of them, but the next person to update hotkeys will have to deal with the same problem.

### CODE REVIEW

**Comment by tortellinitony** [comment]
I'm fine with new emotes, but don't mess with the skin.dmf

**Comment by Spacedong** [comment]
cool

**Comment by cacogen** [comment]
why are you afraid of skin.dmf?

**Review by ghost — CHANGES_REQUESTED** [review]
No.

Please don't fuck with the skin.dmf
Those spawns/sleeps are also bad.

**Comment by cacogen** [comment]
I'm not gonna not fuck with the skin.dmf.

I'm not sure what to say, I don't know why you think that file should be off-limits. I assume you don't understand it.

I added shortcuts to /tg/station years ago using it. It's very straightforward. I've only modified it a handful of times here. The only reason there's so many edits is because the file hasn't been updated in so long and BYOND has changed the formatting. What you're basically saying is that nobody can ever add anymore shortcuts to the game because the file formatting needs to be updated by the latest version of BYOND to do so.

I've tested it without any problems.

**Comment by ghost** [comment]
> I'm not sure what to say, I don't know why you think that file should be off-limits. I assume you don't understand it.

Players don't like change. I also quite frankly don't see the need to put a flex or dance emote down as a hotkey, but others might differ

**Comment by cacogen** [comment]
I knew you'd mention the sleep()/spawn() thing. This was how the code came and all the existing emotes use them. But I'll look at that messy-looking unintuitive CALLBACK() thing.

The only hotkeys I changed were already duplicates. Otherwise I added emotes using alt instead of ctrl. I was thinking of making all the emotes use alt, but I decided to just use ctrl where possible as we already have emotes on ctrl and that's where Goon has them.

For example, both ctrl+Y and Y with hotkeys activates in-hand object. I changed ctrl+Y to an emote.

And yeah it's not necessary to add something like flex or dance to a shortcut because they aren't typing-free forms of communication like nodding or raising your hand or screaming but most of the typing based emotes don't get used and there's no reason not to add them as shortcuts. Ideally I'd like to add a lot more.

**Comment by ghost** [comment]
> I knew you'd mention the sleep()/spawn() thing. This was how the code came and all the existing emotes use them. But I'll look at that messy-looking unintuitive CALLBACK() thing.

Who said we need sleep/spawn? If other emotes use it it's fine, but i quite frankly don't see the need to sleep() inside a emote?

**Comment by JohnGinnane** [comment]
I believe the sleep is for the dance moved

**Comment by ghost** [comment]
> I believe the sleep is for the dance moved

This code pre-dates animate, so this should not be needed if you use parallel animations ;-)

**Comment by j-mie** [comment]
build this

File: code/modules/mob/living/emote.dm [hunk 1]
```diff
@@ -88,7 +88,59 @@
 	key = "dance"
 	key_third_person = "dances"
 	message = "dances around happily."
-	restraint_check = TRUE
```
Comment (tortellinitony) [inline code/modules/mob/living/emote.dm:91]: People shouldn't be able to dance when buckled....

**Review by tortellinitony — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/emote.dm (hunk 1 above)
Comment (cacogen) [reply code/modules/mob/living/emote.dm:91]: if they're buckled it sends the message:  "twitches feebly in time to music only they can hear."

**Comment by cacogen** [comment]
i don't really care enough to keep working on this

**Comment by ghost** [comment]
I'll implement your changes myself then, thanks for the PR anyway.

**Pull request closed without merging**